code/game/objects/objs.dm

```diff
@@ -40,8 +40,11 @@
 
 /obj/Initialize()
 	. = ..()
-	if (!armor)
-		armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
+	if (islist(armor))
+		armor = getArmor(arglist(armor))
+	else if (!istype(armor, /datum/armor))

```

Comment:
can this `if` ever happen?

Reply:
yes, armor is currently defined as a list, it's a "wrong" type for the var but caught here and replaced with the datum.

Reply:
I mean, it already checks for list above, and don't know how it wouldn't be either list or the /datum/armor(maybe a bad map var-edit?)

Reply:
Oh, that if. I thought you meant the islist one. This is trying to catch a scenario where the subtype's own initialize set the armor already, it doesn't happen currently but easily could.